test/integration/multicluster/multicluster-traffic/mc_traffic_test.go

```diff
@@ -0,0 +1,369 @@
+				testutil.AnnotatedFatalf(t,
+					fmt.Sprintf("failed to wait rollout of deploy/%s", "nginx-statefulset"),
+					"failed to wait for rollout of deploy/%s: %s: %s\nEvents:\n%s", "nginx-statefulset", err, o, oEvt)
+			}
+		})
+
+		dgCmd := []string{"--context=" + targetCtx, "diagnostics", "proxy-metrics", "--namespace",
+			"linkerd-multicluster", "deploy/linkerd-gateway"}
+		t.Run("expect open outbound TCP connection from gateway to nginx", func(t *testing.T) {
+			// Use a short time window so that slow-cooker can warm-up and send
+			// requests.
+			err := TestHelper.RetryFor(1*time.Minute, func() error {
+				// Check gateway metrics
+				metrics, err := TestHelper.LinkerdRun(dgCmd...)
+				if err != nil {
+					return fmt.Errorf("failed to get metrics for gateway deployment: %w", err)
+				}
+
+				// If no match, it means there are no open tcp conns from gateway to
+				// nginx pod.
+				if !tcpConnRE.MatchString(metrics) {
+					return fmt.Errorf("failed to find expected TCP connection open outbound metric from gateway to nginx")
+				}
+
+				return nil
+			})
+
+			if err != nil {
+				testutil.AnnotatedFatalf(t, "unexpected error", "unexpected error: %v", err)
+			}
+
+		})
+
+		t.Run("expect non-empty HTTP request metric from gateway to nginx", func(t *testing.T) {
+			// Use a short time window so that slow-cooker can warm-up and send
+			// requests.
+			err := TestHelper.RetryFor(30*time.Second, func() error {
+				// Check gateway metrics
+				metrics, err := TestHelper.LinkerdRun(dgCmd...)
+				if err != nil {
+					return fmt.Errorf("failed to get metrics for gateway deployment: %w", err)
+				}
+
+				// If no match, it means there are no outbound HTTP requests from
+				// gateway to nginx pod.
+				if !httpReqRE.MatchString(metrics) {
+					return fmt.Errorf("failed to find expected outbound HTTP request metric from gateway to nginx\nexpected: %s, got: %s", httpReqRE, metrics)
+				}
+				return nil
+			})
+
+			if err != nil {
+				testutil.AnnotatedFatalf(t, "unexpected error", "unexpected error: %v", err)
+			}
+		})
+	})
+}
+
+func TestSourceResourcesAreCleaned(t *testing.T) {
+	if err := TestHelper.SwitchContext(contexts["src"]); err != nil {
+		testutil.AnnotatedFatalf(t, "failed to rebuild helper clientset with new context", "failed to rebuild helper clientset with new context [%s]: %v", contexts["src"], err)
+	}
+
+	ctx := context.Background()
+	if err := TestHelper.DeleteNamespaceIfExists(ctx, "linkerd-multicluster-statefulset"); err != nil {
+		testutil.AnnotatedFatalf(t, fmt.Sprintf("failed to delete %s namespace", "linkerd-multicluster-statefulset"),
+			"failed to delete %s namespace: %s", "linkerd-multicluster-statefulset", err)
+	}
+
+	if err := TestHelper.DeleteNamespaceIfExists(ctx, "linkerd-emojivoto"); err != nil {
+		testutil.AnnotatedFatalf(t, fmt.Sprintf("failed to delete %s namespace", "linkerd-emojivoto"),
+			"failed to delete %s namespace: %s", "linkerd-emojivoto", err)
+	}
```

Comment:
So these namespaces will exist because they are currently created through the test manifests. If we used `WithDataPlaneNamespace` for these instead they should be deleted upon test completion above.

Reply:
Yeah that's right, but I think that introduces a bit more complexity. Unfortunately the test helper wasn't really built with multiple clientsets and contexts in mind. Until we get to a point where we can effortlessly switch contexts programatically (without having to re-build the whole clientset), I think it's more intuitive to use `WithDataPlaneNamespace` for the context in which we operate. Any side effects (such as creating namespaces and manifests in a different context) can be dealt with after the test finishes imo.

Happy to nest the calls but I think that's going to make this test a bit more confusing and a bit slower. A final note, the first two functions that check the gateway rely on an nginx deployment. Since both tests rely on it, we need to clean-up the resource at the end, or combine the two tests into one (so we can have a `WithDataPlaneNamespace` call

Reply:
Yea so now that these namespaces are no longer in the manifests I think we should be okay removing these. Obviously they don't hurt to have, but I don't think they'd ever find these namespaces now as they aren't being created.